crates/kubelet/src/pod.rs

```diff
@@ -84,6 +84,26 @@ impl Pod {
             .unwrap_or_else(|| &EMPTY_MAP)
     }
 
+    /// Indicate if this pod is a static pod.
+    /// TODO: A missing owner_references field was an indication of static pod in my testing but I
+    /// dont know how reliable this is.
+    pub fn is_static(&self) -> bool {
+        self.0.meta().owner_references.is_none()
+    }
+
+    /// Indicate if this pod is part of a Daemonset
+    pub fn is_daemonset(&self) -> bool {
+        if let Some(owners) = &self.0.meta().owner_references {
+            for owner in owners {
+                match owner.kind.as_ref() {
```

Comment:
nit: use `if let` instead of match here

Reply:
Just curious, what makes one better than the other?

Reply:
It’s just a bit more terse when only one pattern is being handled. It’s the more idiomatic thing to do.

Reply:
It’s just a bit more terse when only one pattern is being handled. It’s the more idiomatic thing to do.

Reply:
It’s just a bit more terse when only one pattern is being handled. It’s the more idiomatic thing to do.